domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotterModule.kt

```diff
@@ -0,0 +1,23 @@
+package org.oppia.android.domain.oppialogger.analytics
+
+import dagger.Module
+import dagger.Provides
+import java.util.concurrent.TimeUnit
+
+/** Provides dependencies that are needed for logging cpu usage. */
+@Module
+class CpuPerformanceSnapshotterModule {
+
+  @Provides
+  fun providesCpuPerformanceSnapshotter(
+    factory: CpuPerformanceSnapshotter.Factory
+  ): CpuPerformanceSnapshotter = factory.createSnapshotter()
```

Comment:
Re: factory, Yeah, I too was on the fence about it -- removed it. 
Re: scoping, I think it'll be better to scope it since it'll be better to only have one instance of the snapshotter that logs metrics sequentially. Otherwise, we might end up with multiple snapshotter instances with their own logging. 
Re: creating a snapshotter class every time we use it -- it's a good idea but won't we need to specifically create this class every time we want to use it? Like yes, ideally it'll only be the observer class for now but this won't be scalable later on -- like in cases where we might want to log cpu usage in different places in the app.